workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityCommandSenderReceiverGenerator.tt

```diff
@@ -286,6 +289,14 @@ namespace <#= qualifiedNamespace #>
                 callbackSystem.RegisterCommandResponseCallback(requestId, wrappedCallback);
             }
         }
+
+        public async Task<<#= receivedCommandResponseType #>> Send<#= commandDetails.CommandName #>Command(<#= commandRequest #> request)
```

Comment:
Usually an API will provide a <Function>Async variant that returns the `Task<>` and allows users the ability to decide if they `await` or not, e.g. `SendCommandNameAsync`

Reply:
I think this does this. If you call this without an await, you will just get a task object and continue synchronously (although it will deadlock if you call call `task.Result`. But that is a function of the task being run in the same thread rather than this function).

Agreed this should be called `SendCommandNameAsyc`.